alacritty/src/event.rs

```diff
@@ -803,6 +807,9 @@ impl<'a, N: Notify + 'a, T: EventListener> input::ActionContext<T> for ActionCon
             self.cancel_search();
         }
 
+        // We don't need IME in Vi mode.
+        self.window().set_ime_allowed(was_in_vi_mode);
```

Comment:
Is it correct to enable IME whenever leaving vi mode? Or should we just disable it if we're entering and it is enabled?

Reply:
It's otherwise you won't have IME when leaving it?

Reply:
Yeah, but are there scenarios where we leave vi mode but still don't want to allow IME? Because otherwise you're forcefully reenabling it here, right? I can't think of anything right now though, so I'd assume we **always** want IME enabled while VI mode isn't enabled?

Reply:
There's no. We use IME by default when in out side of Vi mode.

If you look closely, we start with `IME` enabled, then if we enter `Vi` we disable it. In search mode we just enable it and disable when going back. So it's like consistent. Whenever we can type we enable IME, otherwise(Vi) disable.